source/CIGameScene.cpp

```diff
@@ -87,6 +87,21 @@ bool GameScene::init(const std::shared_ptr<cugl::AssetManager>& assets,
     // create the win scene
     _winScene = WinScene::alloc(assets, dimen);
 
+    // create the pause menu
+    _pauseMenu = PauseMenu::alloc(_assets, networkMessageManager, playerSettings);
+    _pauseMenu->setDisplay(false);
+
+    _pauseBtn = std::dynamic_pointer_cast<scene2::Button>(assets->get<scene2::SceneNode>("game_pausebutton"));
+    _pauseBtn->setColor(Color4::GRAY);
+    _pauseBtn->setVisible(true);
+    _pauseBtn->activate();
+
+    _pauseBtn->addListener([&](const std::string& name, bool down) {
+        if (!down) {
+            _networkMessageManager->setGameState(GameState::GamePaused);
```

Comment:
can you not just call toggle pause here?

Reply:
NVM, we can't edit the properties directly in its listener otherwise it throws an error

Reply:
ok so i think it'd be preferable to add pauseBtn to pauseMenu and then add this listener to the pause menu class with a field indicating the game is paused or not. I don't think we need a game state for paused cuz i thought the game was supposed to just run normally when the game is paused